File: go/internal/rental/rental.go
package rental

import (
	"fmt"
	"time"
)

type Rental struct {
	ID         int
	BikeID     int
	CustomerID int
	StartTime  time.Time
	EndTime    time.Time
}

func NewRental(id int, bikeID int, customerID int) *Rental {
	return &Rental{
		ID:         id,
		BikeID:     bikeID,
		CustomerID: customerID,
		StartTime:  time.Now(),
	}
}

func (r *Rental) CompleteRental() {
	r.EndTime = time.Now()
}

func (r *Rental) IsCompleted() bool {
	return !r.EndTime.IsZero()
}

func (r *Rental) String() string {
	return fmt.Sprintf("Rental ID: %d, bike ID: %d, customer ID: %d, Start time: %s", r.ID, r.BikeID, r.CustomerID, r.StartTime)
}

File: python/modules/rental.py
# Rent bike to customer
##

from modules.bike import Bike
from modules.customer import Customer

class Rental():
    def __init__(self,bikeid: int, customerid: int, startdate,stopdate):
        self._bikeid=bikeid
        self._customerid = customerid
        self._startdate = startdate
        self._stopdate = stopdate
        self._id = -1

    @property
    def bikeid(self):
        return self._bikeid

    @property
    def customerid(self):
        return self._customerid

    @property
    def startdate(self):
        return self._startdate

    @property
    def stopdate(self):
        return self._stopdate

    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, value):
        if value < 0:
            return
        self._id = value